Data/Graph/Inductive/PatriciaTree.hs

```diff
@@ -220,6 +224,25 @@ toAdj = concatMap expand . IM.toList
 fromAdj :: Adj b -> IntMap [b]
 fromAdj = IM.fromListWith addLists . map (second (:[]) . swap)
 
+data FromListCounting a = FromListCounting !Int !(IntMap a)
```

Comment:
I tend to prefer them for when I need to debug something and halfway through a manual application of a function call in ghci I can't get the result out.

I suppose in this case since it's being unwrapped right after it's called it should be OK.